packages/calypso-e2e/src/lib/components/block-widget-editor-component.ts

```diff
@@ -3,9 +3,11 @@ import { Page } from 'playwright';
 const selectors = {
 	editor: '#widgets-editor',
 
-	welcomeModalDismissButton: 'button[aria-label="Close dialog"]',
+	// Welcome Guide and Welcome Tour
+	welcomeModalDismissButton: 'div.components-modal__header > button', // Instead of aria-label, which changes depending on English UK/US.
```

Comment:
This definitely works, but I think there are also a couple of other selectors that would avoid the UK dependency and might be a bit more stable over time.

What do you think of `[aria-label='Welcome to block Widgets'] button[aria-label*=Close]`?

That is specific to this specific modal, avoids hierarchy dependency, and still uses user-facing labels.

Very optional, but wanted to throw out the idea! 😄

Reply:
I think the challenge that this has illustrated is the recommendation by the Playwright team to `prefer front-facing features` as selectors doesn't scale well to an internationalized product.

For instance, I've changed my UI language to Japanese, and the same selector in question then turned into:
```[aria-label="ダイアログを閉じる"]```

Granted, the majority of our E2E tests are executed using `en-US` so this is an non-issue. 
On the other hand, for [MarTech's ToS screenshot project](https://github.com/Automattic/wp-calypso/pull/57741) several selectors already had to be switched back from `aria-label` to a traditional CSS selector due to similar issues.

Ultimately, it's hard to say what's the best approach here. I'm in fact not even sure why the test user in question is occasionally switching to `en-GB` in the first place, but given my experience with `aria-label` so far, I am a bit wary of using it in scenarios where i18n concerns may arise.